src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -3105,7 +3105,10 @@ internal override int CompareKeys(int index1, int index2)
                 if (next == null) return index1 - index2;
                 return next.CompareKeys(index1, index2);
             }
-            return descending ? -c : c;
+            // -c will result in a negative value for int.MaxValue (-int.MinValue == int.MinValue).
```

Comment:
:bulb: This statement is true, but I believe you meant to reference `int.MinValue` where you currently have `int.MaxValue`.